deepinterpolation/inferrence_collection.py

```diff
@@ -233,45 +313,164 @@ def run(self):
         chunk_size = [1]
         chunk_size.extend(self.indiv_shape[:-1])
 
+        output_dataset_name = "data"
+        raw_dataset_name = "raw"
+
         with h5py.File(self.output_file, "w") as file_handle:
-            dset_out = file_handle.create_dataset(
-                "data",
+            file_handle.create_dataset(
+                output_dataset_name,
                 shape=tuple(final_shape),
                 chunks=tuple(chunk_size),
                 dtype=self.output_datatype,
             )
 
-            # Initialize onset of output sample movie
-            local_start = first_sample
+            if self.save_raw:
+                file_handle.create_dataset(
+                    raw_dataset_name,
+                    shape=tuple(final_shape),
+                    chunks=tuple(chunk_size),
+                    dtype=self.output_datatype,
+                )
+
+        logger.info(f"Created empty HDF5 file {self.output_file}")
+        
+        self.model = _load_model(self.json_data)
+
+        # Initialize onset of output sample movie
+        local_start = first_sample
+
+        for epoch_index, index_dataset in enumerate(tqdm(np.arange(self.nb_datasets))):
+            local_data = self.generator_obj.__getitem__(index_dataset)
+
+            # We overwrite epoch_index to allow the last unfilled epoch
+            self.generator_obj.epoch_index = epoch_index
+            local_mean, local_std = \
+                    self.generator_obj.__get_norm_parameters__(index_dataset)  
+            
+            predictions_data = self.model.predict_on_batch(local_data[0])
+            if self.rescale:
+                corrected_data = predictions_data * local_std + local_mean
+            else:
+                corrected_data = predictions_data              
+            local_size = predictions_data.shape[0]
+            start = local_start
+            end = local_start + local_size - 1
+            # We adjust for the next for loop run
+            local_start = end + 1
+
+            start = first_sample + index_dataset * self.batch_size
+            end = first_sample + index_dataset * self.batch_size \
+                + local_size
+    
+            with h5py.File(self.output_file, "a") as file_handle:
+                dset_out = file_handle[output_dataset_name]
+                if self.save_raw:
+                    raw_out = file_handle[raw_dataset_name]
+                    if self.rescale:
+                        corrected_raw = local_data[1] * local_std + local_mean
+                    else:
+                        corrected_raw = local_data[1]
+
+                    raw_out[start:end] = np.squeeze(corrected_raw, -1)
+
+                # We squeeze to remove the feature dimension from tensorflow
+                dset_out[start:end] = np.squeeze(corrected_data, -1)
```

Comment:
looks like this is duplicated code with `write_output_to_file`. Please abstract out duplicated parts so that there is no more duplication